File: src/lib/api/networkPeerClient.ts
import * as pb from "../pb";
import { RPCHost } from "../rpc/host";
import { Readable as GenericReadable } from "../rpc/stream";

export default class NetworkPeer {
  constructor(private readonly host: RPCHost) {}

  public negotiate(
    arg: pb.INetworkPeerNegotiateRequest = new pb.NetworkPeerNegotiateRequest()
  ): Promise<pb.NetworkPeerNegotiateResponse> {
    return this.host.expectOne(
      this.host.call("NetworkPeer/Negotiate", new pb.NetworkPeerNegotiateRequest(arg))
    );
  }
  public open(
    arg: pb.INetworkPeerOpenRequest = new pb.NetworkPeerOpenRequest()
  ): Promise<pb.NetworkPeerOpenResponse> {
    return this.host.expectOne(
      this.host.call("NetworkPeer/Open", new pb.NetworkPeerOpenRequest(arg))
    );
  }
  public close(
    arg: pb.INetworkPeerCloseRequest = new pb.NetworkPeerCloseRequest()
  ): Promise<pb.NetworkPeerCloseResponse> {
    return this.host.expectOne(
      this.host.call("NetworkPeer/Close", new pb.NetworkPeerCloseRequest(arg))
    );
  }
  public updateCertificate(
    arg: pb.INetworkPeerUpdateCertificateRequest = new pb.NetworkPeerUpdateCertificateRequest()
  ): Promise<pb.NetworkPeerUpdateCertificateResponse> {
    return this.host.expectOne(
      this.host.call(
        "NetworkPeer/UpdateCertificate",
        new pb.NetworkPeerUpdateCertificateRequest(arg)
      )
    );
  }
}

File: android/app/src/main/java/gg/strims/ppspp/rpc/NetworkPeerClient.kt
package gg.strims.ppspp.rpc
import gg.strims.ppspp.proto.*

class NetworkPeerClient(filepath: String) : RPCClient(filepath) {

    suspend fun negotiate(
        arg: NetworkPeerNegotiateRequest = NetworkPeerNegotiateRequest()
    ): NetworkPeerNegotiateResponse =
        this.callUnary("NetworkPeer/Negotiate", arg)

    suspend fun open(
        arg: NetworkPeerOpenRequest = NetworkPeerOpenRequest()
    ): NetworkPeerOpenResponse =
        this.callUnary("NetworkPeer/Open", arg)

    suspend fun close(
        arg: NetworkPeerCloseRequest = NetworkPeerCloseRequest()
    ): NetworkPeerCloseResponse =
        this.callUnary("NetworkPeer/Close", arg)

    suspend fun updateCertificate(
        arg: NetworkPeerUpdateCertificateRequest = NetworkPeerUpdateCertificateRequest()
    ): NetworkPeerUpdateCertificateResponse =
        this.callUnary("NetworkPeer/UpdateCertificate", arg)

}
